packages/neon_framework/lib/src/storage/shared_preferences_persistence.dart

```diff
@@ -0,0 +1,110 @@
+// ignore_for_file: cascade_invocations
+
+import 'package:meta/meta.dart';
+import 'package:neon_framework/src/storage/persistence.dart';
+import 'package:shared_preferences_platform_interface/shared_preferences_platform_interface.dart';
+import 'package:shared_preferences_platform_interface/types.dart';
+
+/// An SharedPreferences backed cached persistence for preferences.
+///
+/// There is only one cache backing all `SharedPreferencesPersistence`
+/// instances. Use the [prefix] to separate different storages.
+/// Keys within a storage must be unique.
+@internal
+final class SharedPreferencesPersistence extends CachedPersistence {
+  /// Creates a new SharedPreferences persistence.
+  SharedPreferencesPersistence({String prefix = ''}) : _prefix = prefix;
+
+  static SharedPreferencesStorePlatform get _store => SharedPreferencesStorePlatform.instance;
+
+  /// The prefix of this persistence.
+  ///
+  /// Keys within it must be unique.
+  @protected
+  String get prefix {
+    if (_prefix.isEmpty) {
+      return 'flutter.';
+    }
+
+    return 'flutter.$_prefix-';
+  }
+
+  final String _prefix;
+
+  @override
+  Map<String, Object> get cache {
+    if (_globalCache.containsKey(prefix)) {
+      return _globalCache[prefix]! as Map<String, Object>;
+    }
+
+    final cache = <String, Object>{};
+    final keys = _globalCache.keys;
+    for (final prefixedKey in keys) {
+      if (prefixedKey.startsWith(prefix)) {
+        final key = prefixedKey.replaceFirst(prefix, '');
+        cache[key] = _globalCache[prefixedKey]!;
+      }
+    }
+
+    _globalCache[prefix] = cache;
+    return cache;
```

Comment:
Can you explain why this is needed?

This could end up returning values from storages with a prefix, that clashes with another prefix (e.g. `abc` would also return the values of `abc-def`). Is this intended?

Reply:
I will take a closer look but that's just a limitation of our current shared_prefs implementation.
We can not distinguish between both cases. It doesn't really matter for our current keys but I agree that it is not a good design).
If we only use this persistence for this PR (as a migration) I'd like to just move on with it as it will be deleted in the next one anyways (we can squash them if needed).

Sadly as we already discussed on matrix we might end up with using it on web platforms to avoid longer load times and worse performance for effectively no benefit. 
I'll re think this in the coming days

Reply:
I think that splitting the values into separate maps does not make any sense if we plan to continue using it on web.
I'll use the old approach of concatenating the two keys and storing it in the single  (global) cache. SharedPrefs just isn't made to handle this kind of data separation.